src/main/java/com/microsoft/sqlserver/jdbc/PerformanceLogCallback.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+/**
+ * Callback interface for publishing performance logs.
+ */
+public interface PerformanceLogCallback {
+
+    /**
+     * Publish a performance log entry.
+     *
+     * @param durationMs      The duration of the operation in milliseconds.
+     * @param activity        The type of activity being logged.
+     * @param exceptionMessage An exception message, if an error occurred.
+     */
+    void publish(long durationMs, PerformanceActivity activity, String exceptionMessage);

```

Comment:
Consider adding throws Exception to the publish method signature. The current design silently swallows callback failures which can mask critical issues like configuration errors, resource exhaustion, or compliance violations. Adding throws enables proper error handling, fail-fast behavior, and integration with monitoring systems for better system reliability.

void publish(.....) throws Exception;